samples/sample_common/src/vaapi_utils_x11.cpp

```diff
@@ -32,104 +32,90 @@ or https://software.intel.com/en-us/media-client-solutions-support.
-                                            &cfgAttrib,
-                                            1,
-                                            &vpp_config_id);
-
-            /* Create a context for VPP pipe */
-            va_res = m_libva.vaCreateContext(m_va_dpy,
-                                             vpp_config_id,
-                                             0,
-                                             0,
-                                             VA_PROGRESSIVE,
-                                             0,
-                                             0,
-                                             &m_contextID);
-        }
-#else
-        if (MFX_ERR_NONE == sts)
-        {
-            va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
-            sts = va_to_mfx_status(va_res);
-        }
-#endif // X11_DRI3_SUPPORT
-        if (MFX_ERR_NONE != sts)
-        {
-            m_x11lib.XCloseDisplay(m_display);
-        }
+        msdk_printf(MSDK_STRING("Failed to open X Display: try to check/set DISPLAY environment variable.\n"));
+        throw std::bad_alloc();
     }
-    catch(std::exception& )
+
+    m_va_dpy = m_vax11lib.vaGetDisplay(m_display);
+    if (!m_va_dpy)
     {
-        sts = MFX_ERR_NOT_INITIALIZED;
+        m_x11lib.XCloseDisplay(m_display);
+        msdk_printf(MSDK_STRING("Failed to get VA Display\n"));
+        throw std::bad_alloc();
     }
 
-    if (MFX_ERR_NONE != sts) throw std::bad_alloc();
+    int major_version = 0, minor_version = 0;
+    VAStatus sts = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
+
+    if (VA_STATUS_SUCCESS != sts)
+    {
+        m_x11lib.XCloseDisplay(m_display);
+        msdk_printf(MSDK_STRING("Failed to initialize VAAPI: %d\n"), sts);
+        throw std::bad_alloc();
+    }
+
+#if !defined(X11_DRI3_SUPPORT)
+    VAConfigAttrib cfgAttrib{};
+    if (VA_STATUS_SUCCESS == sts)
+    {
+        cfgAttrib.type = VAConfigAttribRTFormat;
+        sts = m_libva.vaGetConfigAttributes(
+            m_va_dpy,
+            VAProfileNone, VAEntrypointVideoProc,
+            &cfgAttrib, 1);
+    }
+    if (VA_STATUS_SUCCESS == sts)
+    {
+        sts = m_libva.vaCreateConfig(
+            m_va_dpy,
+            VAProfileNone, VAEntrypointVideoProc,
+            &cfgAttrib, 1,
+            &m_configID);
+    }
+    if (VA_STATUS_SUCCESS == sts)
+    {
+        sts = m_libva.vaCreateContext(
+            m_va_dpy,
+            m_configID, 0, 0, VA_PROGRESSIVE, 0, 0,
+            &m_contextID);
+    }
+    if (VA_STATUS_SUCCESS != sts)
+    {
+        Close();
+        msdk_printf(MSDK_STRING("Failed to initialize VP: %d\n"), sts);
+        throw std::bad_alloc();
+    }
+#endif // X11_DRI3_SUPPORT
 }
 
 X11LibVA::~X11LibVA(void)
 {
-    //release context
+    Close();
+}
+
+void X11LibVA::Close()
+{
     if (m_contextID != VA_INVALID_ID)
     {
         m_libva.vaDestroyContext(m_va_dpy, m_contextID);
```

Comment:
I understand that this is almost destructor code. But those functions returns statuses. Can we add at least some logging here to print errors, if they occur?

Reply:
why not? done. Thought I don't think this will be of much use.